lib/acme_server/crypto.ex

```diff
@@ -1,4 +1,7 @@
 defmodule AcmeServer.Crypto do
+  @type id :: {integer(), integer()}
+
+  @spec sign_csr!(id(), binary(), list()) :: binary() | no_return()
```

Comment:
The `list()` is also overly specific. Using `[String.t]` instead would be better. We could also introduce the `domains` type somewhere, since it is also used in the order map.